jdl/exporters/jhipster-deployment-exporter.ts

```diff
@@ -21,53 +21,55 @@ import path from 'path';
 import { createFolderIfItDoesNotExist, doesFileExist } from '../utils/file-utils.js';
 import DeploymentValidator from '../validators/deployment-validator.js';
 import { GENERATOR_NAME, writeConfigFile } from './export-utils.js';
+import JDLDeployment from '../models/jdl-deployment.js';
+import { YoRCJSONObject } from '../converters/types.js';
 
 /**
  * Exports JDL deployments to .yo-rc.json files in separate folders (based on deployment type).
  * @param deployments the deployments to exporters (key: deployment type, value: JDLDeployment- deployment config).
  * @return object[] exported deployments in their final form.
  */
-export default function exportDeployments(deployments) {
+export default function exportDeployments(deployments?: Record<string, JDLDeployment>): Partial<YoRCJSONObject>[] {
```

Comment:
It's not optional.
```suggestion
export default function exportDeployments(deployments: Record<string, JDLDeployment>): Partial<YoRCJSONObject>[] {
```

Reply:
Depends on how we see that: I don't want to change the behavior (yet), so the parameter is optional, and the check will throw...

Reply:
To me, it's equivalent to java
```
int compute(@NonNull x) {...
```
vs 
```
int compute(x) {
if (x == null) {
thrown new ComputationException("shouldn't be null);
```

Reply:
> Depends on how we see that: I don't want to change the behavior (yet), so the parameter is optional, and the check will throw...

There is no behavior change, only typescript definition.

> To me, it's equivalent to java
> 
> ```
> int compute(@NonNull x) {...
> ```
> 
> vs
> 
> ```
> int compute(x) {
> if (x == null) {
> thrown new ComputationException("shouldn't be null);
> ```

It's not equivalent, typescript does not generate any type guard, it should be done by the user.
Typescript should be treated as just as a documentation since it can be used by javascript.
Typescript should be implemented exactly as it would be implemented in javascript + types.